Pegasus/include/geometry.hpp

```diff
@@ -164,1640 +186,667 @@ namespace geometry {
         }
     }
 
-    template <typename ShapeA, typename ShapeB, typename IntersectionSpecific>
-    class IntersectionQueriesBase {
-    protected:
-        ShapeA const* a;
-        ShapeB const* b;
-        bool initialized = false;
-
-    public:
-        explicit IntersectionQueriesBase(ShapeA const* _a = nullptr, ShapeB const* _b = nullptr)
-        {
-            setShapes(_a, _b);
-        }
-
-        void set(ShapeA const* _a, ShapeB const* _b)
-        {
-            setShapes(_a, _b);
-
-            if (initialized) {
-                IntersectionSpecific::calculate();
-            }
-        }
-
-    private:
-        void setShapes(ShapeA const* _a, ShapeB const* _b)
-        {
-            initialized = _a != nullptr && _b != nullptr;
-            if (initialized) {
-                a = _a;
-                b = _b;
-            }
-        }
-    };
+    // Intersection query computation caches
+    struct CacheBase {};
```

Comment:
It will cause a mess in the destructor if we are going to replace it with the void pointers container.